pyquil/_parser/Quil.g4

```diff
@@ -147,10 +163,54 @@ number              : MINUS? ( realN | imaginaryN | I | PI ) ;
 imaginaryN          : realN I ;
 realN               : FLOAT | INT ;
 
+// Analog control
+
+waveformName        : name (DIVIDE name)? ;
+
+defFrame            : DEFFRAME frame ( COLON frameSpec+ )? ;
+frameSpec           : NEWLINE TAB frameAttr COLON ( expression | STRING ) ;
+frameAttr           : SAMPLERATE | INITIALFREQUENCY | DIRECTION | HARDWAREOBJECT | CENTERFREQUENCY;
+
+defWaveform         : DEFWAVEFORM waveformName ( LPAREN param (COMMA param)* RPAREN )? COLON NEWLINE matrix ;
+defCalibration      : DEFCAL name (LPAREN param ( COMMA param )* RPAREN)? qubitOrFormal+ COLON ( NEWLINE TAB instr )* ;
```

Comment:
`COLON ( NEWLINE TAB instr )* ` allows for empty instruction blocks, is that intended?

Reply:
Good catch -- it's not.